docs/v1/P4Runtime-Spec.mdk

```diff
@@ -558,202 +557,170 @@ caption: "Use-Case: Embedded Plus Two Remote High-Availability Controllers" }
+      `role_id`).
+
+## Default Role
+
+A controller can omit the role message in `MasterArbitrationUpdate`. This
+implies the "default role", which corresponds to "full pipeline access". This
+also implies that a default role has a `role.id` of 0 (default). If using a
+default role, all RPCs from the controller (&eg; `Write`) must set the `role_id`
+to 0.
+
+## Role Config
+
+The `role.config` field in the `MasterArbitrationUpdate` message sent by the
+controller describes the role configuration, &ie; which operations, P4 entities,
+behaviors, etc. are in the scope of a given role. An unset `role.config` implies
+"full pipeline access" as well (similar to the default role explained above).
+
+## Rules for Handling `MasterArbitrationUpdate` Messages Received from Controllers
+
+1. If the `MasterArbitrationUpdate` message is received for the first time (for
+   a newly connected controller):
+
+    1. If `device_id` does not match any of the devices known to the P4Runtime
+       server, the server shall terminate the stream by returning a
+       `FAILED_PRECONDITION` error.
+
+    2. If the `election_id` is already used by another controller for the same
+       (`device_id`, `role_id`), the P4Runtime server shall terminate the stream
+       by returning an `INVALID_ARGUMENT` error.
+
+    3. If the max number of clients for the given (`device_id`, `role_id`)
+       exceeds the supported limit, the P4Runtime server shall terminate the
+       stream by returning a `RESOURCE_EXHAUSTED` error.
 
     4. Otherwise, the controller is added to list of connected controllers for
-    the given (device_id, role_id) and the controller will be notified by
-    sending a StreamMessageResponse message back to it, as explained earlier.
+       the given (`device_id`, `role_id`) and the controller is notified by
+       sending a `StreamMessageResponse` message back to it, as explained
+       earlier.
 
-2. If MasterArbitrationUpdate message is received from an already connected
-controller:
+2. If the `MasterArbitrationUpdate` message is received from an already
+   connected controller:
 
-    5. If the device_id does not match the one already assigned to this stream,
-    the P4Runtime server shall terminate the stream by returning a
-    FAILED_PRECONDITION error.
+    1. If the `device_id` does not match the one already assigned to this
+       stream, the P4Runtime server shall terminate the stream by returning a
+       `FAILED_PRECONDITION` error.
 
-    6. Otherwise, if the role.id matches the current role_id assigned to this stream:
+    2. Otherwise, if the `role.id` matches the current `role_id` assigned to
+       this stream:
 
-        * If the election_id also matches the one assigned to this stream, the
-        server will accept role.config only if this controller is the master. If
-        the controller is not a master, the operation is a NOOP.
+        1. If the `election_id` also matches the one assigned to this stream,
```

Comment:
This is producing nested numbers like
```
1.
    1.
          1.
```

As opposed to conventional style:
```
1.
    a.
         i.
```

or sometimes:
```
1.
    1.1
        1.1.1
```

Reply:
I think the way I defined the list in Madoko is correct because it renders as 1. a. i. in the generated PDF. However in HTML it is 1. 1. 1., and I don't know if there is a knob to control that.
I feel like this is not a big deal because of the indentation, although we can always switch to unordered lists for the inner lists.

Reply:
It's fine.